# Sobre o notebook

A "hora da prática" é uma atividade do curso de Python para Data Science da Alura que contém diversos desafios envolvendo os aprendizados de cada aula.

O notebook "hora da pratica" é um notebook construído com espaço para as células dos desafios propostos no curso, com ele você tem espaço para construir e executar suas soluções

## Aula 1 - Introdução ao Python

### Questão 1

Imprima a frase `Escola de Dados da Alura!`

In [ ]:
print('Escola de Dados da Alura')

Escola de Dados da Alura


### Questão 2
Imprima na tela seu nome e seu sobrenome seguindo a estrutura abaixo:
```
Nome: [seu nome]
Sobrenome: [seu sobrenome]
```


In [ ]:
print('Nome: [Gabriel]')
print('Sobrenome: [Pina]')

Nome: [Gabriel]
Sobrenome: [Pina]


### Questão 3
Imprima o seu primeiro nome letra a letra. Por exemplo, nome é Mirla, então eu obtenho a seguinte saída:
```
M
I
R
L
A
```


In [ ]:
print('G')
print('A')
print('B')
print('R')
print('I')
print('E')
print('L')

G
A
B
R
I
E
L


### Questão 4
Imprima o dia do seu nascimento em formato `dia mês ano`. Os valores de dia e ano não podem estar entre aspas. O formato deve estar como no exemplo, supondo uma data de aniversário dia 28 de fevereiro de 2003:
```
28 fevereiro 2003
```

In [ ]:
print(9, ' março ', 2000)

9  março  2000


### Questão 5
Imprima em um único print o atual ano que você está fazendo esse curso, o valor do ano deve ser um dado numérico. A saída do print deve ser a seguinte:
```
Ano atual: [ano]
```


In [ ]:
print('Ano atual [', 2025,']')

Ano atual [ 2025 ]


## Aula 2 - Manipulando dados no Python

### Coleta e amostragem de dados

#### Questão 1
Crie um programa que solicite à pessoa usuária digitar seu nome, e imprima “Olá, [nome]!”.

In [ ]:
nome = input('Digite o seu nome: ')
print(f'Olá, {nome}!')

Digite o seu nome: Gabriel
Olá, Gabriel!


#### Questão 2
Crie um programa que solicite à pessoa usuária digitar seu nome e idade, e imprima “Olá, [nome], você tem [idade] anos.”.

In [ ]:
nome_1 = input('Digite o seu nome: ')
idade_1 = int(input('Digite a sua idade: '))
print(f'Olá, {nome_1}, você tem {idade_1} anos')

Digite o seu nome: Gabriel
Digite a sua idade: 25
Olá, Gabriel, você tem 25 anos


#### Questão 3
Crie um programa que solicite à pessoa usuária digitar seu nome, idade e altura em metros, e imprima “Olá, [nome], você tem [idade] anos e mede [altura] metros!”.

In [ ]:
nome_2 = input('Digite o seu nome: ')
idade_2 = int(input('Digite a sua idade: '))
altura_2 = float(input('Digite a sua altura: '))

print(f'Olá, {nome_2}, você tem {idade_2} anos e mede {altura_2} metros')

Digite o seu nome: Gabriel
Digite a sua idade: 25
Digite a sua altura: 1.75
Olá, Gabriel, você tem 25 anos e mede 1.75 metros


### Calculadora com operadores

#### Questão 1

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a soma dos dois valores.

In [ ]:
num_1 = float(input('Digite o valor do primeiro numero: '))
num_2 = float(input('Digite o valor do segundo numero: '))
soma = num_1 + num_2

print(f'A soma do {num_1} e {num_2} é {soma}')

Digite o valor do primeiro numero: 15
Digite o valor do segundo numero: 35.7
A soma do 15.0 e 35.7 é 50.7


#### Questão 2
Crie um programa que solicite três valores numéricos à pessoa usuária e imprima a soma dos três valores.

In [ ]:
num_1 = float(input('Digite o valor do primeiro numero: '))
num_2 = float(input('Digite o valor do segundo numero: '))
num_3 = float(input('Digite o valor do terceiro numero: '))

soma = num_1 + num_2 + num_3

print(f'A soma do {num_1}, {num_2} e {num_3} é {soma}')

Digite o valor do primeiro numero: 25
Digite o valor do segundo numero: 11.2
Digite o valor do terceiro numero: 9.6
A soma do 25.0, 11.2 e 9.6 é 45.800000000000004


#### Questão 3
Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a subtração do primeiro pelo o segundo valor.

In [ ]:
num_1 = float(input('Digite o valor do primeiro numero: '))
num_2 = float(input('Digite o valor do segundo numero: '))

subtracao = num_1 - num_2

print(f'A subtração do {num_1} pelo {num_2} é {subtracao}')

Digite o valor do primeiro numero: 10
Digite o valor do segundo numero: 7
A subtração do 10.0 pelo 7.0 é 3.0


#### Questão 4

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a multiplicação dos dois valores.


In [ ]:
num_1 = float(input('Digite o valor do primeiro numero: '))
num_2 = float(input('Digite o valor do segundo numero: '))

multiplicacao = num_1 * num_2

print(f'A multiplicação do {num_1} pelo {num_2} é {multiplicacao}')

Digite o valor do primeiro numero: 5
Digite o valor do segundo numero: 9
A multiplicação do 5.0 pelo 9.0 é 45.0


#### Questão 5

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e realize a divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.


In [ ]:
numerador = float(input('Digite o valor do numerador: '))
denominador = float(input('Digite o valor do denominador: '))

divisao = numerador / denominador

print(f'A divisão do {numerador} pelo {denominador} é {divisao}')

Digite o valor do numerador: 50
Digite o valor do denominador: 10
A divisão do 50.0 pelo 10.0 é 5.0


#### Questão 6

Crie um programa que solicite dois valores numéricos, um operador e uma potência, e realize a exponenciação entre esses dois valores.


In [ ]:
operador = float(input('Digite o valor do operador: '))
potencia = float(input('Digite o valor do potência: '))

exponenciacao = operador ** potencia

print(f'A exponenciação do {operador} pelo {potencia} é {exponenciacao}')

Digite o valor do operador: 5
Digite o valor do potência: 3
A exponenciação do 5.0 pelo 3.0 é 125.0


#### Questão 7
Crie um programa que solicite dois valores numéricos, um numerador e um denominador e realize a divisão inteira entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

In [ ]:
numerador = int(input('Digite o valor do numerador: '))
denominador = int(input('Digite o valor do denominador: '))

divisao_inteira = numerador // denominador

print(f'A divisão inteira do {numerador} pelo {denominador} é {divisao_inteira}')

Digite o valor do numerador: 10
Digite o valor do denominador: 3
A divisão inteira do 10 pelo 3 é 3


#### Questão 8

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e retorne o resto da divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

In [ ]:
numerador = float(input('Digite o valor do numerador: '))
denominador = float(input('Digite o valor do denominador: '))

resto_divisao = numerador % denominador

print(f'O resto da divisão do {numerador} pelo {denominador} é {resto_divisao}')

Digite o valor do numerador: 10
Digite o valor do denominador: 3
O resto da divisão do 10.0 pelo 3.0 é 1.0


#### Questão 9

Crie um código que solicita 3 notas de um estudante e imprima a média das notas.

In [ ]:
nota_1 = float(input('Digite a primeira nota: '))
nota_2 = float(input('Digite a segunda nota: '))
nota_3 = float(input('Digite a terceira nota: '))

media = (nota_1 + nota_2 + nota_3) / 3

print(f'A média das notas é {media}')

Digite a primeira nota: 10
Digite a segunda nota: 8
Digite a terceira nota: 5.5
A média das notas é 7.833333333333333


#### Questão 10

Crie um código que calcule e imprima a média ponderada dos números 5, 12, 20 e 15 com pesos respectivamente iguais a 1, 2, 3 e 4.

In [ ]:
n1 = 5
n2 = 12
n3 = 20
n4 = 15

p1 = 1
p2 = 2
p3 = 3
p4 = 4

media_ponderada = (n1 * p1 + n2 * p2 + n3 * p3 + n4 * p4) / (p1 + p2 + p3 + p4)

print(f'A média ponderada é {media_ponderada}')

A média ponderada é 14.9


### Editando textos

#### Questão 1

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase na tela.

In [ ]:
frase = 'Há males que vêm para o bem'
print(frase)

Há males que vêm para o bem


#### Questão 2
Crie um código que solicite uma frase e depois imprima a frase na tela.

In [ ]:
frase = input('Digite uma frase: ')
print(frase)

Digite uma frase: Há males que vêm para o bem
Há males que vêm para o bem


#### Questão 3

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras maiúsculas.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.upper())

Digite uma frase: Há males que vêm para o bem
HÁ MALES QUE VÊM PARA O BEM


#### Questão 4

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras minúsculas.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.lower())

Digite uma frase: HÁ MALES QUE VÊM PARA O BEM
há males que vêm para o bem


#### Questão 5

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase sem espaços em branco no início e no fim.

In [ ]:
frase = '           HÁ MALES QUE VÊM PARA O BEM               '
print(frase.strip())

HÁ MALES QUE VÊM PARA O BEM


#### Questão 6

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.strip())

Digite uma frase:            HÁ MALES QUE VÊM PARA O BEM               
HÁ MALES QUE VÊM PARA O BEM


#### Questão 7


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim e em letras minúsculas.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.strip().lower())

Digite uma frase:            HÁ MALES QUE VÊM PARA O BEM               
há males que vêm para o bem


#### Questão 8

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`e`” trocadas pela letra “`f`”.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.replace('e', 'f'))


Digite uma frase: Gabriel é uma pessoa muito legal
Gabrifl é uma pfssoa muito lfgal


#### Questão 9


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`a`” trocadas pela caractere  “`@`”.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.replace('a', '@'))

Digite uma frase: Gabriel é uma pessoa muito legal
G@briel é um@ pesso@ muito leg@l


#### Questão 10


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as consoantes “`s`” trocadas pelo caractere  “`$`”.

In [ ]:
frase = input('Digite uma frase: ')
print(frase.lower().replace('s', '$'))

Digite uma frase: Sabrina é uma pessoa super simpática
Sabrina é uma pe$$oa $uper $impática


## Aula 03 - Estruturas condicionais

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça à pessoa usuária para fornecer dois números e exibir o número maior.

In [ ]:
num_1 = int(input('Digite o valor do 1° numero: '))
num_2 = int(input('Digite o valor do 2° numero: '))

if num_1 > num_2:
  print(f'O 1° numero ({num_1}) é o maior que o 2° ({num_2})')

elif num_2 > num_1:
  print(f'O 2° numero ({num_2}) é o maior que o 1° ({num_1})')

else:
  print('Os numeros são iguais!')

Digite o valor do 1° numero: 2
Digite o valor do 2° numero: 0
O 1° numero (2) é o maior que o 2° (0)


#### Questão 2
Escreva um programa que solicite o percentual de crescimento de produção de uma empresa e informe se houve um crescimento (porcentagem positiva) ou decrescimento (porcentagem negativa).

In [ ]:
percentual = float(input("Digite o percentual de crescimento da produção (%): "))

if percentual > 0:
  print("Houve crescimento na produção.")

elif percentual < 0:
  print("Houve decrescimento na produção.")

else:
  print('A produção se manteve estável.')

Digite o percentual de crescimento da produção (%): -9.1
Houve decrescimento na produção.


#### Questão 3
Escreva um programa que determine se uma letra fornecida pela pessoa usuária é uma vogal ou consoante.

In [ ]:
vogais = 'aeiou'

letra = input('Digite uma letra: ').lower()

if letra in vogais:
  print(f'A letra {letra} é uma VOGAL!')

else:
  print(f'A letra {letra} é uma CONSOANTE!')

Digite uma letra: O
A letra o é uma VOGAL!


#### Questão 4
Escreva um programa que leia valores médios de preços de um modelo de carro por 3 anos consecutivos e exiba o valor mais alto e mais baixo entre esses três anos.

In [ ]:
vl_carro_1 = float(input('Digite o valor do carro no 1° ano (R$): '))
vl_carro_2 = float(input('Digite o valor do carro no 2° ano (R$): '))
vl_carro_3 = float(input('Digite o valor do carro no 3° ano (R$): '))

if vl_carro_1 > vl_carro_2 > vl_carro_3:
  print(f'O valor mais alto foi de R$ {vl_carro_1} no 1° ano!')
  print(f'O valor mais baixo foi de R$ {vl_carro_3} no 3° ano!')

elif vl_carro_1 > vl_carro_3 > vl_carro_2:
  print(f'O valor mais alto foi de R$ {vl_carro_1} no 1° ano!')
  print(f'O valor mais baixo foi de R$ {vl_carro_2} no 2° ano!')

elif vl_carro_2 > vl_carro_1 > vl_carro_3:
  print(f'O valor mais alto foi de R$ {vl_carro_2} no 2° ano!')
  print(f'O valor mais baixo foi de R$ {vl_carro_3} no 3° ano!')

elif vl_carro_2 > vl_carro_3 > vl_carro_1:
  print(f'O valor mais alto foi de R$ {vl_carro_2} no 2° ano!')
  print(f'O valor mais baixo foi de R$ {vl_carro_1} no 1° ano!')

elif vl_carro_3 > vl_carro_2 > vl_carro_1:
  print(f'O valor mais alto foi de R$ {vl_carro_3} no 3° ano!')
  print(f'O valor mais baixo foi de R$ {vl_carro_1} no 1° ano!')

elif vl_carro_3 > vl_carro_1 > vl_carro_2:
  print(f'O valor mais alto foi de R$ {vl_carro_3} no 3° ano!')
  print(f'O valor mais baixo foi de R$ {vl_carro_2} no 2° ano!')

else:
  print('O carro manteve o valor nos 3 anos!')

Digite o valor do carro no 1° ano (R$): 9800
Digite o valor do carro no 2° ano (R$): 12000
Digite o valor do carro no 3° ano (R$): 25000
O valor mais alto foi de R$ 25000.0 no 3° ano!
O valor mais baixo foi de R$ 9800.0 no 1° ano!


#### Questão 5
Escreva um programa que pergunte sobre o preço de três produtos e indique qual é o produto mais barato para comprar.

In [ ]:
vl_prod_1 = float(input('Digite o valor do 1° produto (R$): '))
vl_prod_2 = float(input('Digite o valor do 2° produto (R$): '))
vl_prod_3 = float(input('Digite o valor do 3° produto (R$): '))

if (vl_prod_1 > vl_prod_2 > vl_prod_3) or (vl_prod_2 > vl_prod_1 > vl_prod_3):
  print(f'O 3° produto é o mais barato a se comprar, R$ {vl_prod_3}')

elif (vl_prod_1 > vl_prod_3) > vl_prod_2 or (vl_prod_3 > vl_prod_1 > vl_prod_2):
  print(f'O 2° produto é o mais barato a se comprar, R$ {vl_prod_2}')

elif (vl_prod_2 > vl_prod_3 > vl_prod_1) or (vl_prod_3 > vl_prod_2 > vl_prod_1):
  print(f'O 1° produto é o mais barato a se comprar, R$ {vl_prod_1}')

else:
  print('Os produtos possuem o mesmo valor!')

Digite o valor do 1° produto (R$): 500
Digite o valor do 2° produto (R$): 61
Digite o valor do 3° produto (R$): 900
O 2° produto é o mais barato a se comprar, R$ 61.0


#### Questão 6
Escreva um programa que leia três números e os exiba em ordem decrescente.

In [ ]:
num_1 = int(input('Digite o valor do 1° numero: '))
num_2 = int(input('Digite o valor do 2° numero: '))
num_3 = int(input('Digite o valor do 3° numero: '))

if num_1 > num_2 > num_3:
  print(f'Ordem decrescente: {num_1}, {num_2}, {num_3}')

elif num_1 > num_3 > num_2:
  print(f'Ordem decrescente: {num_1}, {num_3}, {num_2}')

elif num_2 > num_1 > num_3:
  print(f'Ordem decrescente: {num_2}, {num_1}, {num_3}')

elif num_2 > num_3 > num_1:
  print(f'Ordem decrescente: {num_2}, {num_3}, {num_1}')

elif num_3 > num_1 > num_2:
  print(f'Ordem decrescente: {num_3}, {num_1}, {num_2}')

elif num_3 > num_2 > num_1:
  print(f'Ordem decrescente: {num_3}, {num_2}, {num_1}')

else:
  print('Os 3 numeros digitados possuem o mesmo valor, impossivel ordenar')

Digite o valor do 1° numero: 10
Digite o valor do 2° numero: 90
Digite o valor do 3° numero: 120
Ordem decrescente: 120, 90, 10


#### Questão 7
Escreva um programa que pergunte em qual turno a pessoa usuária estuda ("manhã", "tarde" ou "noite") e exiba a mensagem "Bom Dia!", "Boa Tarde!", "Boa Noite!", ou "Valor Inválido!", conforme o caso.

In [ ]:
nome = input ('Digite o seu nome: ')
turno = input('Você estuda em qual turno? (manhã, tarde ou noite): ').lower()

if turno == 'manhã':
  print(f'Bom dia, {nome}!')

elif turno == 'tarde':
  print(f'Boa tarde, {nome}!')

elif turno == 'noite':
  print(f'Boa noite, {nome}!')

else:
  print(f'Valor do turno inválido, favor tentar novamente, {nome}.')

Digite o seu nome: Benicio
Você estuda em qual turno? (manhã, tarde ou noite): Não sei
Valor do turno inválido, favor tentar novamente, Benicio.


#### Questão 8
Escreva um programa que peça um número inteiro à pessoa usuária e determine se ele é par ou ímpar. Dica: Você pode utilizar o operador módulo `%`.

In [ ]:
numero = int(input('Digite um numero inteiro: '))

if numero % 2 == 0:
  print(f'O numero {numero} é PAR!')

else:
  print(f'O numero {numero} é ÍMPAR!')

Digite um numero inteiro: 7
O numero 7 é ÍMPAR!


#### Questão 9
Escreva um programa que peça um número à pessoa usuária e informe se ele é inteiro ou decimal.

In [ ]:
numero = float(input('Digite um numero: '))

if numero % 1 == 0:
  print(f'O numero {numero} é INTEIRO')

else:
  print(f'O numero {numero} é DECIMAL')

Digite um numero: 25.2
O numero 25.2 é DECIMAL


### Momento dos projetos

#### Questão 10
Um programa deve ser escrito para ler dois números e, em seguida, perguntar à pessoa usuária qual operação ele deseja realizar. O resultado da operação deve incluir informações sobre o número - se é par ou ímpar, positivo ou negativo e inteiro ou decimal.

In [ ]:
num_1 = float(input('Digite o valor do 1° número: '))
num_2 = float(input('Digite o valor do 2° número: '))

opcao = int(input(
    'Opções:\n'
    '(1). Soma \n'
    '(2). Subtração \n'
    '(3). Multiplicação \n'
    '(4). Divisão \n'
    '(5). Exponenciação \n'
    '(6). Média \n'
    'Digite o que precisa ser feito: '
))

# Calcula o resultado e o nome da operação
if opcao == 1:
    resultado = num_1 + num_2
    nome = 'soma'
elif opcao == 2:
    resultado = num_1 - num_2
    nome = 'subtração'
elif opcao == 3:
    resultado = num_1 * num_2
    nome = 'multiplicação'
elif opcao == 4:
    if num_2 == 0:
        print('Erro: divisão por zero não é permitida.')
        exit()
    resultado = num_1 / num_2
    nome = 'divisão'
elif opcao == 5:
    resultado = num_1 ** num_2   # continue usando ** (está ok!)
    nome = 'exponenciação'
elif opcao == 6:
    resultado = (num_1 + num_2) / 2
    nome = 'média'
else:
    print('Opção inválida, tente novamente depois!')
    exit()

print(f'O resultado da {nome} entre {num_1} e {num_2} é {resultado}')

# ---- Classificações ----
# Inteiro ou decimal
eh_inteiro = (resultado % 1 == 0)

# Par/ímpar (só se for inteiro)
if eh_inteiro:
    if resultado % 2 == 0:
        print('PAR')
    else:
        print('ÍMPAR')
else:
    print('DECIMAL (não é par/ímpar)')

# Positivo, negativo ou zero
if resultado > 0:
    print('POSITIVO')
elif resultado < 0:
    print('NEGATIVO')
else:
    print('ZERO (neutro)')

# Reforço do tipo numérico
if eh_inteiro:
    print('INTEIRO')
else:
    print('DECIMAL')


Digite o valor do 1° número: 10
Digite o valor do 2° número: 2
Opções:
(1). Soma 
(2). Subtração 
(3). Multiplicação 
(4). Divisão 
(5). Exponenciação 
(6). Média 
Digite o que precisa ser feito: 4
O resultado da divisão entre 10.0 e 2.0 é 5.0
ÍMPAR
POSITIVO
INTEIRO


#### Questão 11
Escreva um programa que peça à pessoa usuária três números que representam os lados de um triângulo. O programa deve informar se os valores podem ser utilizados para formar um triângulo e, caso afirmativo, se ele é equilátero, isósceles ou escaleno. Tenha em mente algumas dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;


In [ ]:
num_1 = float(input('Digite o valor 1° lado: '))
num_2 = float(input('Digite o valor 2° lado: '))
num_3 = float(input('Digite o valor 3° lado: '))

if (num_1 + num_2 > num_3) and (num_1 + num_3 > num_2) and (num_2 + num_3 > num_1):
  if (num_1 == num_2 and num_2 == num_3):
    print('Triângulo Equilátero')

  elif (num_1 == num_2 or num_1 == num_3 or num_2 == num_3):
    print('Triângulo Isósceles')

  else:
    print('Triângulo Escaleno')

else:
   print('Os valores informados não formam um triângulo')

Digite o valor 1° lado: 10
Digite o valor 2° lado: 8
Digite o valor 3° lado: 6
Triângulo Escaleno


#### Questão 12
Um estabelecimento está vendendo combustíveis com descontos variados. Para o etanol, se a quantidade comprada for até 15 litros, o desconto será de 2% por litro. Caso contrário, será de 4% por litro. Para o diesel, se a quantidade comprada for até 15 litros, o desconto será de 3% por litro. Caso contrário, será de 5% por litro. O preço do litro de diesel é R\$ 2,00 e o preço do litro de etanol é R\$ 1,70. Escreva um programa que leia a quantidade de litros vendidos e o tipo de combustível (E para etanol e D para diesel) e calcule o valor a ser pago pelo cliente. Tenha em mente algumas dicas:

- O do valor do desconto será a multiplicação entre preço do litro, quantidade de litros e o valor do desconto.
- O valor a ser pago por um cliente será o resultado da multiplicação do preço do litro pela quantidade de litros menos o valor de desconto resultante do cálculo.


In [ ]:
litros = float(input('Digite a quantidade de litros: '))
combustivel = input('Digite o desejado: \n'
                    '(E) Etanol \n'
                    '(D) Diesel\n'
                  ).upper()

if (combustivel == 'E'):  # Valor do Etanol R$1,70 o litro
  if (litros <= 15):
    # desconto 2% por litro
    valor_total = (litros * 1.70) * (1 - 0.02)
    print(f'Para {litros} litros de Etanol, irá custar R$ {valor_total:.2f}')
  else:
    # desconto 4% por litro
    valor_total = (litros * 1.70) * (1 - 0.04)
    print(f'Para {litros} litros de Etanol, irá custar R$ {valor_total:.2f}')

elif (combustivel == 'D'):  # Valor do Diesel R$2,00 o litro
  if (litros <= 15):
    # desconto 3% por litro
    valor_total = (litros * 2.00) * (1 - 0.03)
    print(f'Para {litros} litros de Diesel, irá custar R$ {valor_total:.2f}')
  else:
    # desconto 5% por litro
    valor_total = (litros * 2.00) * (1 - 0.05)
    print(f'Para {litros} litros de Diesel, irá custar R$ {valor_total:.2f}')

else:
  print('Valor incorreto')


Digite a quantidade de litros: 10
Digite o desejado: 
(E) Etanol 
(D) Diesel
E
Para 10.0 litros de Etanol, irá custar R$ 16.66


#### Questão 13
Em uma empresa de venda de imóveis você precisa criar um código que analise os dados de vendas anuais para ajudar a diretoria na tomada de decisão. O código precisa coletar os dados de quantidade de venda durante os anos de 2022 e 2023 e fazer um cálculo de variação percentual. A partir do valor da variação, deve ser enviada às seguintes sugestões:
- Para variação acima de 20%: bonificação para o time de vendas.
- Para variação entre 2% e 20%: pequena bonificação para time de vendas.
- Para variação entre 2% e -10%: planejamento de políticas de incentivo às vendas.
- Para bonificações abaixo de -10%: corte de gastos.


In [ ]:
vendas_22 = float(input('Quantidade de vendas em 2022: '))
vendas_23 = float(input('Quantidade de vendas em 2023: '))

variacao = ((vendas_23 - vendas_22) / vendas_22) * 100

if (variacao > 20):
  print(f'A variação foi de {variacao}\nBONIFICAÇÃO PARA O TIME DE VENDAS!')

elif (2 < variacao <= 20):
  print(f'A variação foi de {variacao}\nPEQUENA BONIFICAÇÃO PARA O TIME DE VENDAS!')

elif (-10 < variacao <= 2):
  print(f'A variação foi de {variacao}\nPLANEJAMENTO DE POLÍTICAS DE INCENTIVO ÁS VENDAS!')

else:
  print(f'A variação foi de {variacao}\nCORTE DE GASTOS!')

Quantidade de vendas em 2022: 100
Quantidade de vendas em 2023: 115
A variação foi de 15.0
PEQUENA BONIFICAÇÃO PARA O TIME DE VENDAS!


## Aula 4 - Estruturas de repetição

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça dois números inteiros e imprima todos os números inteiros entre eles.

In [ ]:
for i in range(1, 3):
  numero = int(input('Digite um numero inteiro: '))
  print(f'Numero digitado: {numero}')

Digite um numero inteiro: 10
Numero digitado: 10
Digite um numero inteiro: 5
Numero digitado: 5


#### Questão 2
Escreva um programa para calcular quantos dias levará para a colônia de uma bactéria A ultrapassar ou igualar a colônia de uma bactéria B, com base nas taxas de crescimento de 3% e 1,5% respectivamente. Considere que a colônia A inicia com 4 elementos e a colônia B com 10 elementos.

In [ ]:
# Populações iniciais
colonia_A = 4
colonia_B = 10

# Taxas de crescimento
taxa_A = 0.03  # 3%
taxa_B = 0.015 # 1,5%

dias = 0

# Simulação diária
while colonia_A < colonia_B:
    colonia_A += colonia_A * taxa_A
    colonia_B += colonia_B * taxa_B
    dias += 1

print(f"A colônia A ultrapassará ou igualará a colônia B em {dias} dias.")
print(f"Tamanho final da colônia A: {colonia_A:.2f}")
print(f"Tamanho final da colônia B: {colonia_B:.2f}")


A colônia A ultrapassará ou igualará a colônia B em 63 dias.
Tamanho final da colônia A: 25.75
Tamanho final da colônia B: 25.55


#### Questão 3
Para tratar uma quantidade de 15 dados de avaliações de pessoas usuárias de um serviço da empresa, precisamos verificar se as notas são válidas. Então, escreva um programa que vai receber a nota de 0 a 5 de todos os dados e verificar se é um valor válido. Caso seja inserido uma nota acima de 5 ou abaixo de 0, repita até que a pessoa usuária insira um valor válido.

In [ ]:
notas = []  # lista para guardar as 15 notas válidas

# Vamos repetir 15 vezes
for i in range(15):
    while True:  # loop infinito até digitar uma nota válida
        nota = float(input(f"Digite a nota {i+1} (entre 0 e 5): "))

        if 0 <= nota <= 5:  # verifica se está no intervalo válido
            notas.append(nota)  # guarda a nota
            break  # sai do while e vai para a próxima nota
        else:
            print("Nota inválida! Digite um valor entre 0 e 5.")

# Mostrar todas as notas válidas no final
print("\nAs notas válidas inseridas foram:")
print(notas)


Digite a nota 1 (entre 0 e 5): -1
Nota inválida! Digite um valor entre 0 e 5.
Digite a nota 1 (entre 0 e 5): 10
Nota inválida! Digite um valor entre 0 e 5.
Digite a nota 1 (entre 0 e 5): 5
Digite a nota 2 (entre 0 e 5): 3
Digite a nota 3 (entre 0 e 5): 2
Digite a nota 4 (entre 0 e 5): 1
Digite a nota 5 (entre 0 e 5): 5
Digite a nota 6 (entre 0 e 5): 6
Nota inválida! Digite um valor entre 0 e 5.
Digite a nota 6 (entre 0 e 5): 7
Nota inválida! Digite um valor entre 0 e 5.
Digite a nota 6 (entre 0 e 5): 2
Digite a nota 7 (entre 0 e 5): 1
Digite a nota 8 (entre 0 e 5): 3
Digite a nota 9 (entre 0 e 5): 4
Digite a nota 10 (entre 0 e 5): 2
Digite a nota 11 (entre 0 e 5): 1
Digite a nota 12 (entre 0 e 5): 2
Digite a nota 13 (entre 0 e 5): 3
Digite a nota 14 (entre 0 e 5): 4
Digite a nota 15 (entre 0 e 5): 5

As notas válidas inseridas foram:
[5.0, 3.0, 2.0, 1.0, 5.0, 2.0, 1.0, 3.0, 4.0, 2.0, 1.0, 2.0, 3.0, 4.0, 5.0]


#### Questão 4
Desenvolva um programa que leia um conjunto indeterminado de temperaturas em Celsius e faça uma análise. Portanto, escreva um programa que leia temperaturas e informe a média delas. A leitura deve ser encerrada ao ser enviado o valor -273°C.

In [ ]:
lista_temp = []

while True:
    temp = int(input('Digite uma temperatura (Celsius): '))
    if temp == -273:  # condição de parada
        break
    lista_temp.append(temp)

# só calcula a média se a lista não estiver vazia
if lista_temp:
    media_temp = sum(lista_temp) / len(lista_temp)
    print(f'A média de temperaturas é de {media_temp:.2f} °C')
else:
    print("Nenhuma temperatura válida foi inserida.")


Digite uma temperatura (Celsius): 10
Digite uma temperatura (Celsius): 50
Digite uma temperatura (Celsius): -10
Digite uma temperatura (Celsius): 30
Digite uma temperatura (Celsius): 25
Digite uma temperatura (Celsius): 11
Digite uma temperatura (Celsius): 12
Digite uma temperatura (Celsius): 31
Digite uma temperatura (Celsius): -273
A média de temperaturas é de 19.88 °C


#### Questão 5
Escreva um programa que calcule o fatorial de um número inteiro fornecido pela pessoa usuária. O fatorial de um número inteiro é a multiplicação desse número por todos os seus antecessores até o número 1. Por exemplo, o fatorial de 5 é 5 x 4 x 3 x 2 x 1 = 120.

In [ ]:
numero = int(input('Digite um número inteiro: '))

resultado = 1
i = 1
while i <= numero:
    resultado *= i
    i += 1

print(f"O fatorial de {numero} é {resultado}")


Digite um número inteiro: 10
O fatorial de 10 é 3628800


### Momento dos projetos

#### Questão 6
Escreva um programa que gere a tabuada de um número inteiro de 1 a 10, de acordo com a escolha da pessoa usuária. Como exemplo, para o número 2, a tabuada deve ser mostrada no seguinte formato:
```
Tabuada do 2:
2 x 1 = 2
2 x 2 = 4
[...]
2 x 10 = 20
```

In [ ]:
numero = int(input('Digite um numero inteiro de 1 a 10: '))

for i in range(1, 11):
  mult = i * numero
  print(f'{numero} x {i} = {mult}')

Digite um numero inteiro de 1 a 10: 2
2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18
2 x 10 = 20


#### Questão 7
Os números primos possuem várias aplicações dentro da Ciência de Dados, por exemplo, na criptografia e segurança. Um número primo é aquele que é divisível apenas por um e por ele mesmo. Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.

In [ ]:
numero = int(input("Digite um número: "))

if numero <= 1:
    print("Não é primo.")
else:
    primo = True
    for i in range(2, numero):  # testa divisores de 2 até n-1
        if numero % i == 0:
            primo = False
            break
    if primo:
        print("É primo.")
    else:
        print("Não é primo.")


Digite um número: 13
É primo.


#### Questão 8
Vamos entender a distribuição de idades de pensionistas de uma empresa de previdência. Escreva um programa que leia as idades de uma quantidade não informada de clientes e mostre a distribuição em intervalos de [0-25], [26-50], [51-75] e [76-100]. Encerre a entrada de dados com um número negativo.

In [ ]:
# Inicializa os contadores de cada faixa
faixa_1 = 0  # [0-25]
faixa_2 = 0  # [26-50]
faixa_3 = 0  # [51-75]
faixa_4 = 0  # [76-100]

while True:
    idade = int(input("Digite a idade: "))

    # Condição de parada
    if idade < 0:
        break

    # Verifica em qual faixa a idade se encaixa
    if 0 <= idade <= 25:
        faixa_1 += 1
    elif 26 <= idade <= 50:
        faixa_2 += 1
    elif 51 <= idade <= 75:
        faixa_3 += 1
    elif 76 <= idade <= 100:
        faixa_4 += 1
    else:
        print("Idade fora do intervalo válido (0-100). Não será contabilizada.")

# Mostra os resultados
print("\nDistribuição de idades:")
print(f"[0-25]: {faixa_1}")
print(f"[26-50]: {faixa_2}")
print(f"[51-75]: {faixa_3}")
print(f"[76-100]: {faixa_4}")


Digite a idade: 50
Digite a idade: 21
Digite a idade: 31
Digite a idade: 11
Digite a idade: 0
Digite a idade: 36
Digite a idade: 100
Digite a idade: 150
Idade fora do intervalo válido (0-100). Não será contabilizada.
Digite a idade: 21
Digite a idade: 32
Digite a idade: 33
Digite a idade: 41
Digite a idade: 65
Digite a idade: 12
Digite a idade: 11
Digite a idade: 10
Digite a idade: 5
Digite a idade: 9
Digite a idade: -1

Distribuição de idades:
[0-25]: 9
[26-50]: 6
[51-75]: 1
[76-100]: 1


#### Questão 9
Em uma eleição para gerência em uma empresa com 20 funcionários, existem quatro candidatos. Escreva um programa que calcule o vencedor da eleição. A votação ocorreu da seguinte maneira:
- Cada funcionário votou em um dos quatro candidatos (representados pelos números 1, 2, 3 e 4).
- Também foram contabilizados os votos nulos (representado pelo número 5) e os votos em branco (representado pelo número 6).

Ao final da votação, o programa deve exibir o total de votos para cada candidato, o número de votos nulos e o número de votos em branco. Além disso, deve calcular e exibir a porcentagem de votos nulos em relação ao total de votos e a porcentagem de votos em branco em relação ao total de votos.

In [ ]:
candidato_1 = 0
candidato_2 = 0
candidato_3 = 0
candidato_4 = 0
nulo_5 = 0
branco_6 = 0

# Total de eleitores
total_votos = 20

for i in range(1, total_votos + 1):
    numero_cand = int(input(f'Digite o número do candidato (1 a 6) - Voto {i}: '))

    if numero_cand == 1:
        candidato_1 += 1
    elif numero_cand == 2:
        candidato_2 += 1
    elif numero_cand == 3:
        candidato_3 += 1
    elif numero_cand == 4:
        candidato_4 += 1
    elif numero_cand == 5:
        nulo_5 += 1
    elif numero_cand == 6:
        branco_6 += 1
    else:
        print('Número inválido! Voto não contabilizado.')

# Exibir resultados
print("\nResultado da eleição:")
print(f"Candidato 1: {candidato_1} votos")
print(f"Candidato 2: {candidato_2} votos")
print(f"Candidato 3: {candidato_3} votos")
print(f"Candidato 4: {candidato_4} votos")
print(f"Votos Nulos: {nulo_5}")
print(f"Votos em Branco: {branco_6}")

# Calcular porcentagens
porcentagem_nulos = (nulo_5 / total_votos) * 100
porcentagem_brancos = (branco_6 / total_votos) * 100

print(f"\nPorcentagem de votos Nulos: {porcentagem_nulos:.2f}%")
print(f"Porcentagem de votos em Branco: {porcentagem_brancos:.2f}%")

# Determinar vencedor
votos_candidatos = {
    "Candidato 1": candidato_1,
    "Candidato 2": candidato_2,
    "Candidato 3": candidato_3,
    "Candidato 4": candidato_4
}

vencedor = max(votos_candidatos, key=votos_candidatos.get)
print(f"\nO vencedor da eleição foi: {vencedor} com {votos_candidatos[vencedor]} votos!")
22

Digite o número do candidato (1 a 6) - Voto 1: 1
Digite o número do candidato (1 a 6) - Voto 2: 2
Digite o número do candidato (1 a 6) - Voto 3: 3
Digite o número do candidato (1 a 6) - Voto 4: 4
Digite o número do candidato (1 a 6) - Voto 5: 5
Digite o número do candidato (1 a 6) - Voto 6: 5
Digite o número do candidato (1 a 6) - Voto 7: 5
Digite o número do candidato (1 a 6) - Voto 8: 5
Digite o número do candidato (1 a 6) - Voto 9: 5
Digite o número do candidato (1 a 6) - Voto 10: 6
Digite o número do candidato (1 a 6) - Voto 11: 6
Digite o número do candidato (1 a 6) - Voto 12: 6
Digite o número do candidato (1 a 6) - Voto 13: 3
Digite o número do candidato (1 a 6) - Voto 14: 2
Digite o número do candidato (1 a 6) - Voto 15: 3
Digite o número do candidato (1 a 6) - Voto 16: 3
Digite o número do candidato (1 a 6) - Voto 17: 2
Digite o número do candidato (1 a 6) - Voto 18: 1
Digite o número do candidato (1 a 6) - Voto 19: 4
Digite o número do candidato (1 a 6) - Voto 20: 5

Resultad

22

## Aula 5 - Estruturas de dados

### Aquecendo na programação

#### Questão 1
Faça um programa que tenha a seguinte lista contendo os valores de gastos de uma empresa de papel `[2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]`. Com esses valores, faça um programa que calcule a média de gastos. Dica, use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).

In [ ]:
lista_gastos = [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]

media = sum(lista_gastos) / len(lista_gastos)

print(f'A média gasta pela empresa de papel foi de R$ {media:.2f}')

A média gasta pela empresa de papel foi de R$ 3144.22


#### Questão 2
 Com os mesmos dados da questão anterior, defina quantas compras foram acima de 3000 reais e calcule a porcentagem quanto ao total de compras.

In [ ]:
quantidade = 0
valores_acima = []

for valor in lista_gastos:
  if valor > 3000:
    quantidade += 1
    valores_acima.append(valor)

porcentagem = (quantidade / len(lista_gastos) * 100)

print(f'A quantidade de acima de R$ 3.000,00 foi de {quantidade}.')
print(f'Isso representa {porcentagem:.2f}% do total de compras.')
print(f'Os valores foram: {valores_acima}')


A quantidade de acima de R$ 3.000,00 foi de 7.
Isso representa 70.00% do total de compras.
Os valores foram: [3701.35, 3518.09, 3456.61, 3249.38, 3891.45, 3075.26, 3219.08]


#### Questão 3
Faça um código que colete em uma lista 5 números inteiros quaisquer e imprima a lista. Exemplo: `[1,4,7,2,4]`.

In [ ]:
lista_numeros = []

for i in range (1, 6):
  numero = int(input(f'Digite o {i}° numero: '))
  lista_numeros.append(numero)

print(lista_numeros)

Digite o 1° numero: 1
Digite o 2° numero: 3
Digite o 3° numero: 6
Digite o 4° numero: 7
Digite o 5° numero: 9
[1, 3, 6, 7, 9]


#### Questão 4
Colete novamente 5 inteiros e imprima a lista em ordem inversa à enviada.

In [ ]:
lista_numeros = []

for i in range (1, 6):
  numero = int(input(f'Digite o {i}° numero: '))
  lista_numeros.append(numero)

lista_numeros.reverse()
print(lista_numeros)

Digite o 1° numero: 1
Digite o 2° numero: 2
Digite o 3° numero: 3
Digite o 4° numero: 4
Digite o 5° numero: 5
[5, 4, 3, 2, 1]


#### Questão 5
Faça um programa que, ao inserir um número qualquer, criará uma lista contendo todos os números primos entre 1 e o número digitado.

In [ ]:
num_primos = []
lim_primo = int(input('Digite um número: '))

for num in range(2, lim_primo + 1):
    primo = True
    for i in range(2, num):  # testa divisores de 2 até num-1
        if num % i == 0:
            primo = False
            break
    if primo:
        num_primos.append(num)

print(f"Números primos entre 1 e {lim_primo}: {num_primos}")


Digite um número: 15
Números primos entre 1 e 15: [2, 3, 5, 7, 11, 13]


#### Questão 6
Escreva um programa que peça uma data informando o dia, mês e ano e determine se ela é uma data válida para uma análise.

In [ ]:
# função para verificar se um ano é bissexto
def bissexto(ano):
    if (ano % 4 == 0 and ano % 100 != 0) or (ano % 400 == 0):
        return True
    return False

# entrada de dados
dia = int(input("Digite o dia: "))
mes = int(input("Digite o mês: "))
ano = int(input("Digite o ano: "))

# meses com 31 dias
meses_31 = [1, 3, 5, 7, 8, 10, 12]
# meses com 30 dias
meses_30 = [4, 6, 9, 11]

valida = True  # assumimos que é válido e vamos checar

# valida mês
if mes < 1 or mes > 12:
    valida = False
else:
    # verifica dias de acordo com o mês
    if mes in meses_31 and not (1 <= dia <= 31):
        valida = False
    elif mes in meses_30 and not (1 <= dia <= 30):
        valida = False
    elif mes == 2:
        if bissexto(ano) and not (1 <= dia <= 29):
            valida = False
        elif not bissexto(ano) and not (1 <= dia <= 28):
            valida = False

# resultado
if valida:
    print(f"A data {dia:02d}/{mes:02d}/{ano} é válida.")
else:
    print("Data inválida!")


Digite o dia: 35
Digite o mês: 03
Digite o ano: 2025
Data inválida!


### Momento dos projetos

#### Questão 7
Para um estudo envolvendo o nível de multiplicação de bactérias em uma colônia, foi coletado o número de bactérias multiplicadas por dia, com base em um processo de duplicação diária e pode ser observado a seguir: `[1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]`. Tendo esses valores, faça um código que gere uma lista contendo o percentual de crescimento de bactérias por dia, comparando o número de bactérias em cada dia com o número de bactérias do dia anterior. Dica: para calcular o percentual de crescimento usamos a seguinte equação: `100 * (amostra_atual - amostra_passada) / (amostra_passada)`.

In [ ]:
perc_bacteria = []
mult_bacteria = [1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]

for i in range(1, len(mult_bacteria)):
    amostra_passada = mult_bacteria[i - 1]
    amostra_atual = mult_bacteria[i]
    percentual = 100 * (amostra_atual - amostra_passada) / amostra_passada
    perc_bacteria.append(round(percentual, 2))  # arredonda para 2 casas

print(f"Segue a evolução de contaminação em porcentagem:\n {perc_bacteria}")


Segue a evolução de contaminação em porcentagem:
 [75.0, 57.14, 51.52, 56.0, 44.87, 46.9, 51.2, 50.6, 50.53]


#### Questão 8
Para uma seleção de produtos alimentícios, precisamos separar o conjunto de IDs dados por números inteiros, sabendo que os produtos com ID par são doces e os com ID ímpar são amargos. Monte um código que colete 10 IDs. Depois, calcule e mostre a quantidade de produtos doces e amargos.

In [ ]:
lista_doce = []
lista_amargos = []

for i in range(10):
    id_produto = int(input("Digite o ID do produto: "))
    if id_produto % 2 == 0:
        lista_doce.append(id_produto)
    else:
        lista_amargos.append(id_produto)

print(f"A quantidade de doces: {len(lista_doce)}")
print(f"A quantidade de amargos: {len(lista_amargos)}")


Digite o ID do produto: 10
Digite o ID do produto: 25
Digite o ID do produto: 34
Digite o ID do produto: 11
Digite o ID do produto: 35
Digite o ID do produto: 19
Digite o ID do produto: 6
Digite o ID do produto: 7
Digite o ID do produto: 8
Digite o ID do produto: 28
A quantidade de doces: 5
A quantidade de amargos: 5


#### Questão 9
Desenvolva um programa que informa a nota de um aluno de acordo com suas respostas. Ele deve pedir a resposta de um aluno para cada questão e é preciso verificar se a resposta foi igual ao gabarito. Cada questão vale um ponto e existem alternativas:  A, B, C ou D.
```
Gabarito da Prova:
01 - D
02 - A
03 - C
04 - B
05 - A
06 - D
07 - C
08 - C
09 - A
10 - B
```

In [ ]:
gabarito = ['D', 'A', 'C', 'B', 'A', 'D', 'C', 'C', 'A', 'B']
acertos = 0

for cont in range(len(gabarito)):
  resposta = input(f'Qual a alternativa da {cont+1}° questão? ').upper()
  if resposta == gabarito[cont]:
    acertos += 1
  else:
    acertos += 0
print(f'Nota do aluno {acertos}/{len(gabarito)}')


Qual a alternativa da 1° questão? D
Qual a alternativa da 2° questão? A
Qual a alternativa da 3° questão? C
Qual a alternativa da 4° questão? A
Qual a alternativa da 5° questão? A
Qual a alternativa da 6° questão? C
Qual a alternativa da 7° questão? D
Qual a alternativa da 8° questão? D
Qual a alternativa da 9° questão? A
Qual a alternativa da 10° questão? B
Nota do aluno 6/10


#### Questão 10
Um instituto de meteorologia deseja fazer um estudo de temperatura média de cada mês do ano. Para isso, você precisa fazer um código que colete e armazene essas temperaturas médias em uma lista. Depois, calcule a média anual das temperaturas e mostre todas as temperaturas acima da média anual, e em que mês elas ocorreram, mostrando os meses por extenso: Janeiro, Fevereiro, etc.

In [ ]:
meses = ["Janeiro", "Fevereiro", "Março", "Abril", "Maio", "Junho",
         "Julho", "Agosto", "Setembro", "Outubro", "Novembro", "Dezembro"]

temperaturas = {}

for mes in meses:
    temperaturas[mes] = float(input(f"Digite a média de temperatura de {mes}: "))

media_temp = sum(temperaturas.values()) / len(temperaturas)

print(f"\nMédia anual de temperatura: {media_temp:.2f}°C")
print("Meses acima da média anual:")

for mes, temp in temperaturas.items():
    if temp > media_temp:
        print(f"{mes}: {temp}°C")


Digite a média de temperatura de Janeiro: 37
Digite a média de temperatura de Fevereiro: 36.5
Digite a média de temperatura de Março: 25
Digite a média de temperatura de Abril: 22
Digite a média de temperatura de Maio: 30
Digite a média de temperatura de Junho: 21
Digite a média de temperatura de Julho: 18
Digite a média de temperatura de Agosto: 20
Digite a média de temperatura de Setembro: 23
Digite a média de temperatura de Outubro: 16
Digite a média de temperatura de Novembro: 20
Digite a média de temperatura de Dezembro: 21

Média anual de temperatura: 24.12°C
Meses acima da média anual:
Janeiro: 37.0°C
Fevereiro: 36.5°C
Março: 25.0°C
Maio: 30.0°C


#### Questão 11
Uma empresa de e-commerce está interessada em analisar as vendas dos seus produtos. Os dados das vendas foram armazenados em um dicionário:
```
{'Produto A': 300, 'Produto B': 80, 'Produto C': 60,
 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}
```
Escreva um código que calcule o total de vendas e o produto mais vendido.

In [ ]:
produtos = {
    'Produto A': 300,
    'Produto B': 80,
    'Produto C': 60,
    'Produto D': 200,
    'Produto E': 250,
    'Produto F': 30
}

# total de vendas
total_vendas = sum(produtos.values())

# produto mais vendido
mais_vendido = max(produtos, key=produtos.get)

print(f"Total de vendas: {total_vendas}")
print(f"Produto mais vendido: {mais_vendido} ({produtos[mais_vendido]} unidades)")


Total de vendas: 920
Produto mais vendido: Produto A (300 unidades)


#### Questão 12
Uma pesquisa de mercado foi feita para decidir qual design de uma marca infantil mais agrada crianças. A pesquisa foi feita e o votos computados podem ser observados abaixo:
```
'''
Tabela de votos da marca
Design 1 - 1334 votos
Design 2 - 982 votos
Design 3 - 1751 votos
Design 4 - 210 votos
Design 5 - 1811 votos
'''
```
Adapte os dados fornecidos a você para uma estrutura de dicionário e a partir dele, informe o design vencedor e a porcentagem de votos recebidos.


In [ ]:
design = {
    'Design_1' : 1334,
    'Design_2' : 982,
    'Design_3' : 1751,
    'Design_4' : 210,
    'Design_5' : 1811
}

vencedor = max(design, key=design.get)
total_votos = sum(design.values())
perc_votos = round((design[vencedor] / total_votos) * 100, 2)

print(f'O Design vencedor foi o {vencedor} com {perc_votos}% dos votos!')


O Design vencedor foi o Design_5 com 29.75% dos votos!


#### Questão 13
Os funcionários de um setor da empresa que você trabalha vão receber um abono correspondente a 10% do seu salário devido ao ótimo desempenho do time. O setor financeiro solicitou sua ajuda para a verificação das consequências financeiras que esse abono irá gerar nos recursos. Assim, foi encaminhada para você uma lista com os salários que receberão o abono: `[1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]`. O abono de cada funcionário não pode ser inferior a 200. Em código, transforme cada um dos salários em chaves de um dicionário e o abono de cada salário no elemento. Depois, informe o total de gastos com o abono, quantos funcionários receberam o abono mínimo e qual o maior valor de abono fornecido.


In [ ]:
salarios = [1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]

abonos = {}  # dicionário: salário → abono
for s in salarios:
    valor_abono = round(s * 0.10, 2)
    if valor_abono < 200:
        valor_abono = 200
    abonos[s] = valor_abono

# total gasto
total_gasto = sum(abonos.values())

# quantidade de funcionários que receberam o abono mínimo
qtd_minimo = list(abonos.values()).count(200)

# maior valor de abono
maior_abono = max(abonos.values())

print("Dicionário salário → abono:", abonos)
print(f"Total gasto com abonos: R$ {total_gasto:.2f}")
print(f"Funcionários que receberam o abono mínimo: {qtd_minimo}")
print(f"Maior valor de abono fornecido: R$ {maior_abono:.2f}")


Dicionário salário → abono: {1172: 200, 1644: 200, 2617: 261.7, 5130: 513.0, 5532: 553.2, 6341: 634.1, 6650: 665.0, 7238: 723.8, 7685: 768.5, 7782: 778.2, 7903: 790.3}
Total gasto com abonos: R$ 6087.80
Funcionários que receberam o abono mínimo: 2
Maior valor de abono fornecido: R$ 790.30


#### Questão 14
Uma equipe de cientistas de dados está estudando a diversidade biológica em uma floresta. A equipe fez a coleta de informações sobre o número de espécies de plantas e animais em cada área da floresta e armazenaram essas informações em um dicionário. Nele, a chave descreve a área dos dados e os valores nas listas correspondem às espécies de plantas e animais nas áreas, respectivamente.
```
{'Área Norte': [2819, 7236],
 'Área Leste': [1440, 9492],
 'Área Sul': [5969, 7496],
 'Área Oeste': [14446, 49688],
 'Área Centro': [22558, 45148]}
```
Escreva um código para calcular a média de espécies por área e identificar a área com a maior diversidade biológica. Dica: use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).


In [ ]:
areas = {
    'Área Norte': [2819, 7236],
    'Área Leste': [1440, 9492],
    'Área Sul': [5969, 7496],
    'Área Oeste': [14446, 49688],
    'Área Centro': [22558, 45148]
}

medias = {}
totais = {}

for area, especies in areas.items():
    media = sum(especies) / len(especies)   # média entre plantas e animais
    total = sum(especies)                   # total de espécies
    medias[area] = media
    totais[area] = total

# área com maior diversidade
area_mais_diversa = max(totais, key=totais.get)

# resultados
print("Média de espécies por área:")
for area, media in medias.items():
    print(f"{area}: {media:.2f}")

print(f"\nÁrea com maior diversidade biológica: {area_mais_diversa} ({totais[area_mais_diversa]} espécies)")


Média de espécies por área:
Área Norte: 5027.50
Área Leste: 5466.00
Área Sul: 6732.50
Área Oeste: 32067.00
Área Centro: 33853.00

Área com maior diversidade biológica: Área Centro (67706 espécies)


#### Questão 15
O setor de RH da sua empresa te pediu uma ajuda para analisar as idades dos funcionários de 4 setores da empresa. Para isso, ele te forneceu os seguintes dados:
```
{'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}
```
Sabendo que cada setor tem 10 funcionários, construa um código que calcule a média de idade de cada setor, a idade média geral entre todos os setores e quantas pessoas estão acima da idade média geral.


In [ ]:
idades = {
    'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
    'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
    'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
    'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]
}

# média por setor
medias_setores = {}
for setor, lista in idades.items():
    medias_setores[setor] = sum(lista) / len(lista)

# todas as idades juntas
todas_idades = []
for lista in idades.values():
    todas_idades.extend(lista)

media_geral = sum(todas_idades) / len(todas_idades)

# quantas pessoas estão acima da média geral
acima_media = sum(1 for idade in todas_idades if idade > media_geral)

# resultados
print("Média de idade por setor:")
for setor, media in medias_setores.items():
    print(f"{setor}: {media:.2f}")

print(f"\nIdade média geral: {media_geral:.2f}")
print(f"Quantidade de pessoas acima da média geral: {acima_media}")


Média de idade por setor:
Setor A: 39.90
Setor B: 42.30
Setor C: 35.80
Setor D: 36.50

Idade média geral: 38.62
Quantidade de pessoas acima da média geral: 18
